stdlib/src/utils/static_tuple.mojo

```diff
@@ -254,3 +314,139 @@ struct StaticTuple[element_type: AnyRegType, size: Int](Sized):
         var base_ptr = Pointer[Self.type].address_of(self.array).address
         var ptr = __mlir_op.`pop.array.gep`(base_ptr, Int(0).value)
         return Pointer(ptr)
+
+
+# ===----------------------------------------------------------------------===#
+# Array
+# ===----------------------------------------------------------------------===#
+
+
+@value
+struct Array[ElementType: CollectionElement, size: Int](Sized):
+    """A statically sized tuple type which contains elements of homogeneous types.
```

Comment:
**Suggestion** Since we don't have tuple in the name, WDYT about phrasing this something along the lines of 


> A fixed-size sequence of `size` homogenous elements where `size` is a constant expression.

or something like that.

Reply:
I think keeping the feature but marking it private and subject to change with an underscore prefix would be better.  I believe the ability to access the underlying storage of `static_tuple` was requested specifically and only added recently so it would be a shame to lose it.

Also,  UnsafePointers  or References to `_array` are created several times in the code.  `UnsafePointer` now converts to an immortal `Reference` with suffix `[]`.  So `_as_ptr()` and `_as_ptr()[]` could replace all these and  `_get_reference_unsafe` could be removed.  

Since creating an immortal Reference from an UnsafePointer is new maybe others more knowledgable than me need to confirm its use is a good idea.

Reply:
@mikowals Maybe you meant to put your comment in the `as_ptr` thread?